packages/mui-material/src/Accordion/Accordion.js

```diff
@@ -168,7 +168,10 @@ const Accordion = React.forwardRef(function Accordion(inProps, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const backwardCompatibleSlots = { transition: TransitionComponentProp, ...slots };
+  const backwardCompatibleSlots = {
+    transition: slotProps?.transition?.component ?? TransitionComponentProp,
```

Comment:
It's weird to me that we have to do this manually 🤔 
Did you figure out what was causing the details not to show?

Reply:
Sorry for late reply, i was sick for a week and couldn't dig much. i'll figure this out this weekend

Reply:
No problem! I hope you're feeling better 🤗

Reply:
i did some digging, this is what i found.  root cause of the issue because `Collapse` is not capable of handling `<Collapse as={ReactComponent}` it can handle React.ElementType like `<Collapse as='span'/>`.

`  const [TransitionSlot, transitionProps] = useSlot('transition', {`

`TrasnitionSlot` from above code is either `slots.transition` or `elementType` (which is passed in useSlot). If component has `slotProps.transition.component` then this component is present under `as` key in `transitionProps`. and component under `as` key is finally rendered.  Since `Collapse` is not capable of handling `as={ReactComponent}` AccordionSummary is not getting displayed.

To understand this more clearly, you can paste this code in any demos, and observe only  2 `X`'s are displayed. where ideally 3 `X`'s should be displayed.

```
import * as React from 'react';
import Box from '@mui/material/Box';
import Collapse from '@mui/material/Collapse';
import { Fade } from '@mui/material';

export default function SimpleCollapse() {
  return (
    <div>
      <Box sx={{ width: '50%' }}>
        <Collapse as={'span'} orientation="horizontal" in={true}>
          X
        </Collapse>
        <Collapse as={Collapse} orientation="horizontal" in={true}>
          X
        </Collapse>
      </Box>
      <Box sx={{ width: '50%' }}>
        <Fade as={Fade} in={true}>
          <span>X</span>
        </Fade>
      </Box>
    </div>
  );
}
```

Reply:
I understand. This seems to me like a `Collapse` issue, would you agree? I think we should solve it there.